src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -130,6 +131,37 @@ public void execute(final MavenProject project) {
         return new MinimizerPlugin(runner).runDependentTestFile(DetectorPathManager.detectionFile());
     }
 
+    // Determine which minimized result to try and fix
+    private void setupAndApplyFix(final List<MinimizeTestsResult> minimizedResults) throws Exception {
+        // Check that there are only two passed in results
+        if (minimizedResults.size() != 2) {
+            TestPluginPlugin.error("There are " + minimizedResults.size() + " results, should be two");
+            return;
+        }
+
+        // Go through the results, looking specifically for not passing and has polluters
+        for (MinimizeTestsResult minimized : minimizedResults) {
```

Comment:
It seems this part just fixes the first polluter/setter that it finds. I remember discussing this briefly before, but not sure what we concluded (if anything). I think it's probably best to fix all polluters/setters, and I believe the code already checks to make sure that the tests fails as expected before trying to fix it.

Reply:
Actually, this part is iterating over the pair of MinimizeTestsResult (in the form of a List, but it was checked before to have size of 2) for the same dependent test, just checking if it is a case with a polluter or with a setter. Since at this point it can only be one of the two, the return is fine.

Reply:
I think the place to handle multiple polluters/setters is in setupAndApplyFix. We might want to make a separate PR for doing this part.